scripts/uberenv/packages/conduit/package.py

```diff
@@ -0,0 +1,520 @@
+          dir when calling this function. `py_site_pkgs_dir` should
+          be an absolute path or `None`.
+
+          This is necessary because the spack `site_packages_dir`
+          var will not exist in the base class. For more details
+          on this issue see: https://github.com/spack/spack/issues/6261
+        """
+
+        #######################
+        # Compiler Info
+        #######################
+        c_compiler = env["SPACK_CC"]
+        cpp_compiler = env["SPACK_CXX"]
+        f_compiler = None
+
+        if self.compiler.fc:
+            # even if this is set, it may not exist
+            # do one more sanity check
+            if os.path.isfile(env["SPACK_FC"]):
+                f_compiler  = env["SPACK_FC"]
+
+        #######################################################################
+        # By directly fetching the names of the actual compilers we appear
+        # to doing something evil here, but this is necessary to create a
+        # 'host config' file that works outside of the spack install env.
+        #######################################################################
+
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+
+        # are we on a specific machine
+        on_blueos = 'blueos' in sys_type
+
+        ##############################################
+        # Find and record what CMake is used
+        ##############################################
+
+        cmake_exe = spec['cmake'].command.path
+        host_cfg_fname = "%s-%s-%s-conduit.cmake" % (socket.gethostname(),
+                                                     sys_type,
+                                                     spec.compiler)
+
+        cfg = open(host_cfg_fname, "w")
+        cfg.write("##################################\n")
+        cfg.write("# spack generated host-config\n")
+        cfg.write("##################################\n")
+        cfg.write("# {0}-{1}\n".format(sys_type, spec.compiler))
+        cfg.write("##################################\n\n")
+
+        # Include path to cmake for reference
+        cfg.write("# cmake from spack \n")
+        cfg.write("# cmake executable path: %s\n\n" % cmake_exe)
+
+        #######################
+        # Compiler Settings
+        #######################
+
+        cfg.write("#######\n")
+        cfg.write("# using %s compiler spec\n" % spec.compiler)
+        cfg.write("#######\n\n")
+        cfg.write("# c compiler used by spack\n")
+        cfg.write(cmake_cache_entry("CMAKE_C_COMPILER", c_compiler))
+        cfg.write("# cpp compiler used by spack\n")
+        cfg.write(cmake_cache_entry("CMAKE_CXX_COMPILER", cpp_compiler))
+
+        # SERAC PATCH BEGIN
+        # we need to specify the c++ std
+        cfg.write(cmake_cache_entry("BLT_CXX_STD", "c++11"))
+        # SERAC PATCH END
```

Comment:
@white238 , for my education:
- Why couldn’t this patch be applied in Conduit?
- What is BLT doing specifically with c++11?
- Why wouldn’t conduit add the exact standard feature requirement so that CMake can verify compiler compatibility and apply the necessary flags? (cmake compile features)

Reply:
I guess the answer to the second one is: RTFM...

Reply:
1) It should, but first it needs to be verified that this solves the problem @jamiebramwell is running into
2) CMake has a couple variables it uses to control the C++ std (C++ and CUDA) so this unifies them but it is not strictly required
3) Libraries try to leave standards open as much as possible so they can be toggled by the host code but conduit should (and it is being added currently) default to C++11 if it is not explicitly stated.

Reply:
Yeah, ideally this would go directly in conduit. I think @cyrush is hitting the same issue right now. I'm just working on it in my own space to see if I can help.

Reply:
@white238 after we discussed, I rechecked -- if the cxx standard isn't set -- conduit *does* set it to c++11, here is the logic:

https://github.com/LLNL/conduit/blob/f592a1067af1643e935488579ef5a2ae16da75c3/src/cmake/SetupBLT.cmake#L55

So not sure what is going on? 
If there is a bug in how it is done in the conduit build system, I would like to understand and fix it there.

Reply:
@jamiebramwell hardcoded BLT_CXX_STD as well on the platform/compiler she ran into the problem and it didn't make the build error go away as well.

Reply:
maybe an issue with the BLT I am using with that compiler? Maybe try with a newer external blt?

Reply:
The failure is in Axom build.
However Axom Spack package uses hostconfig files, and the toss3 Spack config does not have Clang9 compiler defined.
Maybe this is where ABI get mixed. Still investigating.

Reply:
Axom compiler config shouldn’t interfere though.

Reply:
There is a distinct possibility no one has tried the conduit->axom workflow with clang9.

Reply:
build commands in conduit have the `-std=cxx11`, but not `--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.1.0`.

Reply:
Added cxx_flags definition in host-config file for conduit, seems to work!

Reply:
did we accidentally remove the writing of flags to the host-configs i put in recently? Though i don't think thats solving jamie's problem since her problem is showing up on WSL2 and doesn't have any global flags

Reply:
I confirmed. I think i gave bad advice and the conduit package has changes that are not in spack proper that need to be restored in our repo

Reply:
Actually now that I think about it, both changes from axom and conduit can now be pushed to spack proper now that a couple PR's in their repo have gone through.  I'll take care of that later...

Reply:
https://github.com/LLNL/serac/commit/cab14bef4579c36c4bdb1fb8b0808ea296b89053#diff-2cfc1f4f3c26f3bb4ba5de09656bde95
This file has a few more fixes in it.